File: 0100. Same Tree/solution.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution
{
    public void WalkTree(TreeNode p, TreeNode q, ref bool same)
    {
        if (p != null && q != null)
        {
            if (p.val != q.val)
                same = false;

            WalkTree(p.left, q.left, ref same);
            WalkTree(p.right, q.right, ref same);
        }
        else if (p != null && q == null || p == null && q != null)
            same = false;
    }

    public bool IsSameTree(TreeNode p, TreeNode q)
    {
        bool same = true;
        WalkTree(p, q, ref same);
        return same;
    }
}

File: 0100. Same Tree/solution.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution 
{
public:
    void walkTree(TreeNode* p, TreeNode* q, bool &same)
    {
        if (p && q) 
        {
            if (p->val != q->val)
            {
                same = false;
                return;
            }

            walkTree(p->right, q->right, same);
            walkTree(p->left, q->left, same);
        }
        else if (p && q == nullptr || q && p == nullptr)
        {
            same = false;
            return;
        }
    }
    
    bool isSameTree(TreeNode* p, TreeNode* q) {
        bool same = true;
        walkTree(p, q, same);
        return same;
    }
};